### ISSUE TITLE
Add context manager for `critical_path` pointcut `around`

### ISSUE THREAD

**Issue by SteVwonder** [flux-framework/PerfFlowAspect#5]
@Jmast suggested that it would be nice to support finer granularity than function-level with a context manager.  As a strawman example:

```python
def myfunc(args):
   # do stuff
   with perfflow.aspect.critical_path():
       # do more stuff
       print("Inside the profiled event")
```

Context mangers are nice for this since they are guaranteed to have both their `__enter__` and `__exit__` methods called even if there is an exception etc.  Note: I don't think we can use the same name for both the decorator and the context manager.

**Comment by dongahn** [comment]
Yes I finally got that when Jeff was making this request. This should be easy enough to support I believe as well.
